src/firebase_token.rs

```diff
@@ -0,0 +1,72 @@
+//! Extracting bearer tokens from request headers
+//! and with the assumption that they're Firebase JWTs, checking
+//! their validity.
+
+use crate::{errors::{Error, InvalidAuthHeader, InvalidJwt}, DecodedToken, FirebaseAuth};
+use rocket::{request::{Request, FromRequest, self}, http::Status};
+
+/// The bearer token included in request headers
+///
+/// Valid requests with the `Authorization` header included should look
+/// like the following:
+/// ```ignore
+/// Authorization: Bearer <some_bearer_token>
+/// ```
+#[derive(Debug)]
+pub struct FirebaseToken {
```

Comment:
I wonder if we can use `DecodedToken` directly instead of declaring a wrapper struct. What do you think?

Reply:
Yes, definitely possible. The reason why I didn't choose that path is because then the guard's name would also be `DecodedToken`, so people would be including `DecodedToken` instead of `FirebaseToken`.

Also, some more information can be included in the wrapper struct for potential future uses. Such as perhaps the user's email, profile picture and other information that can be retrieved from Firebase?

Reply:
I've created a new PR to address the confusing variable name and missing fields for the FirebaseToken, as you pointed out. It should be covered now!
https://github.com/DrPoppyseed/rocket-firebase-auth/pull/8